GitUI/Hotkey/HotkeySettingsManager.cs

```diff
@@ -10,37 +10,47 @@
 
 namespace GitUI.Hotkey
 {
-    internal static class HotkeySettingsManager
+    public interface IHotkeySettingsManager : IHotkeySettingsReader
+{
+        HotkeySettings[] CreateDefaultSettings();
+        bool IsUniqueKey(Keys keyData);
+        HotkeySettings[] LoadSettings();
+        void SaveSettings(HotkeySettings[] settings);
+    }
+
+    internal class HotkeySettingsManager : IHotkeySettingsManager
     {
-        #region Serializer
         private static XmlSerializer? _serializer;
+        private static readonly HashSet<Keys> _usedKeys = new();
+        private readonly IScriptsManager _scriptsManager;
 
         /// <summary>Lazy-loaded Serializer for HotkeySettings[].</summary>
         private static XmlSerializer Serializer => _serializer ??= new XmlSerializer(typeof(HotkeySettings[]), new[] { typeof(HotkeyCommand) });
```

Comment:
You lost me here... why?

Reply:
> `static` should not be intermixed with member fields.

I mistakenly thought this was part of the Internal Coding Guidelines.
Anyway, I strongly prefer to separate static from instance members because this is an important difference.

Regardless of this:

> Have you forgotten to remove `static`?

You have turned this class from a `static` to an instance class.
It is not obvious that `_usedKeys` should remain `static`.

Reply:
The original .NET Framework internal guidelines state that methods should be alphabetised:
![image](https://github.com/gitextensions/gitextensions/assets/4403806/7c8536dd-f542-4f74-998b-18140fcb5364)

You can find the guidelines here: https://learn.microsoft.com/en-us/archive/blogs/brada/internal-coding-guidelines#28-file-organization

. . .

Over the years I tried and followed many different guidelines, but having worked on some large codebases, I found the .NET Framework internal guidelines to work best especially if the code isn't browsed in an IDE. This way I always know that properties are always before methods, and I can easily find the desired member by scrolling the file up or down.
Whether a method is static or not is immaterial, it's just a syntactic sugar for the compiler.

Reply:
> It is not obvious that `_usedKeys` should remain `static`.

Thank you for pointing this out. It shouldn't, I totally missed this.